cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -273,6 +276,31 @@ class alignas(16) column_device_view : public detail::column_device_view_base {
     return data<T>()[element_index];
   }
 
+  /**---------------------------------------------------------------------------*
+   * @brief Iterator for navigating this column
+   *---------------------------------------------------------------------------**/
+  template <typename T>
+  struct value_accessor;
+  using count_it = thrust::counting_iterator<size_type>;
+  template <typename T>
+  using const_iterator = thrust::transform_iterator<value_accessor<T>, count_it>;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Return new iterator pointing to the beginning of this column
+   *---------------------------------------------------------------------------**/
+  template <typename T>
+  const_iterator<T> begin() const {
```

Comment:
Need to document that this is for non-nullable columns only and will throw if nullable.